### ISSUE TITLE
Expose protocol as variable for external tools

### ISSUE THREAD

**Issue by toubsen** [mRemoteNG/mRemoteNG#1634]
Externalize the variable "protocol" for usage in the command line of external tools.

## Expected Behavior
I'd like to be able to open connections of type HTTP + HTTPS in an external browser.

## Current Behavior
 Currently, I need to external tool entries, to differentiate between HTTP and HTTPS, to make entries of the following kind:
- `"http://%Hostname%:%PORT%"`
- `"https://%Hostname%:%PORT%"`

Additionally, I need to know in advance if a certain entry is of type HTTP or HTTPS, to call it with the correct external tool, which (if you don't remember exactly) can be very annoying.

## Possible Solution
With the variable "protocol" I could construct an external tool parameter like the following:
- `"%PROTOCOL%://%Hostname%:%PORT%"`

## Steps to Reproduce (for bugs)
Configure external tool:
1. Display name: "Open in Chrome"
2. Filename: "chrome.exe"
3. Parameter "http://%Hostname%:%PORT%"
4. Cannot open HTTPS URLs now

## Context
Trying to open any connection of type URL with an external browser, without duplicating external tool entries and without needing to peek into the connection config, if it's actually HTTP or HTTPS.

## Your Environment

* Version used: 1.76.20 (2019-04-12):
* Operating System and version: Windows 10 1903 x64 (Build 18362.418)

**Comment by kees-z** [comment]
Agreed, this should be added, but why not expose **ALL** settings to external apps in the first place? I make intensive use of external apps. I regularly run into limitations because a specific setting cannot be passed to an external app.
Any setting could be sent using a predefined format, like `%GROUP__SETTING%`, for example `%PROTOCOL__USE_CREDSSP%`.

**Comment by Kvarkas** [comment]
> Agreed, this should be added, but why not expose **ALL** settings to external apps in the first place? I make intensive use of external apps. I regularly run into limitations because a specific setting cannot be passed to an external app.
> Any setting could be sent using a predefined format, like `%GROUP__SETTING%`, for example `%PROTOCOL__USE_CREDSSP%`.

Combination of all settings can be vey hudge, perhaps need a better idea how to archive such but I got you point

**Comment by robertpopa22** [comment]
Triage refresh (P2): this issue has been inactive for a long time. Please retest on the latest build and share reproducible steps, expected vs actual behavior, and logs/screenshots. If there is no confirmation/update, maintainers can close it as stale or not reproducible.

**Comment by robertpopa22** [comment]
Implemented in PR #3123 from fork branch codex/pr19-protocol-1634. 
 
Scope: 
- Add PROTOCOL token support in ExternalToolArgumentParser 
- Add parser tests for token variants 
 
Validation: 
- Local MSBuild (Release/x64) passed 
- ExternalToolsArgumentParserTests passed (37/37)

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by robertpopa22** [comment]
**Fix available in community release**

This fix is included in a community build from our fork (`robertpopa22/mRemoteNG`, branch `main`). Available from the fork's Releases page.

Please test and let us know if the issue is fully resolved.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
